src/test/java/com/example/SpringMongoProject/Controller/StudentController_getStudents_88425bb65e_Test.java

```diff
@@ -0,0 +1,52 @@
+// Test generated by RoostGPT for test springMongoDB using AI Type Open AI and AI Model gpt-4
+
+package com.example.SpringMongoProject.Controller;
+
+import com.example.SpringMongoProject.Entity.Student;
+import com.example.SpringMongoProject.Service.StudentServices;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.boot.test.context.SpringBootTest;
+import static org.mockito.Mockito.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest
+public class StudentController_getStudents_88425bb65e_Test {
+
+    @InjectMocks
+    StudentController studentController;
+
+    @Mock
+    StudentServices studentServices;
+
+    @BeforeEach
+    public void init() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void testGetStudents_success() {
+        String studentId = "123";
+        Student student = new Student();
+        student.set_id(studentId);
+        when(studentServices.getStudentByID(studentId)).thenReturn(student);
```

Comment:
Similar issue here The error is indicating that when is not applicable for methods with a void return type. Since saveorUpdate is likely a void method, you should use doNothing() to mock its behavior.